server/mqtt.go

```diff
@@ -2624,9 +2623,13 @@ func (as *mqttAccountSessionManager) loadRetainedMessagesForSubject(topSubject [
 			// not happen.
 			continue
 		}
-		if ref.loaded != nil && ref.loaded.expiresFromCache.After(time.Now()) {
```

Comment:
Few lines above, you still reference `as.retmsgs[subject]` outside of the lock, and you can't.

Reply:
I guess if you go back to use `rms`, this will be removed.